parsers/src/main/java/org/semanticweb/owlapi/rdf/syntax/RDFParser.java

```diff
@@ -84,29 +84,45 @@
 @SuppressWarnings("unused")
 public class RDFParser extends DefaultHandler implements RDFConstants {
 
-    protected static final Locator s_nullDocumentLocator = new LocatorImpl();
-    protected static final SAXParserFactory s_parserFactory = buildFactory();
+    private static ThreadLocal<Locator> locatorThreadLocal = new ThreadLocal<Locator>() {
+        @Override
+        protected Locator initialValue() {
+            return new LocatorImpl();
+        }
+    };
+
+    protected static Locator getLocator() {
+        return locatorThreadLocal.get();
+    }
+    private static ThreadLocal<SAXParserFactory> saxParserFactoryThreadLocal = new ThreadLocal<SAXParserFactory>() {
+
+        @Override
+        protected SAXParserFactory initialValue() {
+            try {
+                System.setProperty("entityExpansionLimit", "100000000");
```

Comment:
I've made a different fix in version 4, without the need to set system properties - the values can be set directly on the generated parser. There is a utility method on SaxParsers to centralize this, so that client code does not need to worry.